module-system/module-implementations/sov-evm/src/lib.rs

```diff
@@ -88,14 +88,43 @@ mod experimental {
         pub(crate) block_env: sov_state::StateValue<BlockEnv>,
 
         #[state]
-        pub(crate) transactions: sov_state::StateMap<Bytes32, RawEvmTransaction>,
+        pub(crate) head_number: sov_state::StateValue<u64>,
 
+        // TODO JsonCodec: This is a workaround for https://github.com/bincode-org/bincode/issues/245 which affects all
+        // binary serialization formats.
+        // 1. Implement custom types for Block, Transaction etc.. with borsh derived.
+        // 2. Remove JsonCodec.
         #[state]
-        pub(crate) receipts: sov_state::StateMap<
-            ethereum_types::H256,
-            TransactionReceipt,
-            sov_state::codec::BcsCodec,
+        pub(crate) current_block: sov_state::StateValue<reth_rpc_types::Block, JsonCodec>,
```

Comment:
I think we should keep using `block_env` instead of `current_block` and just create a block in the `end_slot_hook`

Reply:
if we add pending_transactions, then just Header would suffice. Or if we want to accumulate transactions in it than full block (or our own type containing header and transactions)